R/guides-.R

```diff
@@ -529,11 +528,28 @@ Guides <- ggproto(
     grobs <- vector("list", length(guides))
     for (i in seq_along(grobs)) {
       grobs[[i]] <- guides[[i]]$draw(
-        theme = theme, position = as.character(positions[i]),
+        theme = theme, position = positions[i],
         direction = directions[i], params = params[[i]]
       )
+      # we'll merge inside legends with same coordinate into same guide box
+      # here, we define the groups of the inside legends
+      if (identical(positions[i], "inside")) {
+        positions[i] <- paste(
+          "inside",
+          paste(attr(.subset2(grobs, i), "inside_position"), collapse = "_"),
+          sep = "_"
+        )
+      }
```

Comment:
Instead of reading out the attributes here, you can deduce what the inside position is by using outside this loop:

```
default_inside_position <- calc_element("legend.position.inside", theme) %||%
  valid.just(calc_element("legend.justification.inside"))
```
and this inside the loop:
```
calc_element("legend.position.inside", params[[i]]$theme) %||% default_inside_position
```

This would allow us to skip editing guide-colorbar.R and guide-legend.R.

Reply:
Dose this means we should allow user to set numeric value in the `position` argument of `guide_legend` and `guide_colorbar`?

Reply:
No I still think the position string left/right/top/bottom/inside is separate from the position location when legends are placed inside. I imagine you can set the position location with `guide_legend(theme = theme(legend.inside.position)` as in your example.

Reply:
In this case, if we want users to set the inside legend position via the theme argument, we would need to modify the `GuideLegend` or `GuideColorbar` to ensure that the elements include the `legend.inside.position`.

Reply:
The same with the `GuideCustom` class.

Reply:
There was a reason I had to divorce numerical legend positions from string positions, although at the moment I can't quickly recall what that reason was. We do not need to guide to register the `legend.inside.position` as an element if it is not needed to render the guide itself. It is only needed to compose the guides into guide boxes. `position` is an exception, and implemented as a paramter, because axis guides absolutely need to know which side of the plotting panel they're on. Consistency then dictated that it should also be a parameter to legends, colourbars etc.

Reply:
Thanks for the details, the position will now be managed in `Guides$package_box()`.